bot/cogs/source.py

```diff
@@ -16,15 +16,23 @@ class Source(commands.Cog):
     def __init__(self, bot: commands.Bot) -> None:
         self.bot = bot
 
+    @in_channel(constants.Channels.bot)
     @commands.command(name="source")
-    async def command_source(self, ctx: commands.Context, command_name: str = None) -> None:
+    async def command_source(self, ctx: commands.Context, command_name: str = None, sub_command: str = None) -> None:
         """View the source of a command."""
         if command_name is None:
-            return await ctx.send("> https://github.com/python-discord/bot")
+            await ctx.send("> https://github.com/python-discord/bot")
+            return
+
+        if sub_command:
+            await ctx.send(sub_command)
+            command = self.bot.get_command(f"{command_name} {sub_command}")
+        else:
+            command = self.bot.get_command(command_name)
 
-        command = self.bot.get_command(command_name)
```

Comment:
Instead of that `sub_command` usage, we could do `self.bot.get_command(command_name)`.